### ISSUE TITLE
Sending GZIP version for the wrong request header?

### ISSUE THREAD

**Issue by rprieto** [dustyburwell/connect-file-cache#8]
Hi,

I'm running into an issue running [Zombie.js](http://zombie.labnotes.org) on my app, which uses
`Express.js > connect-assets > connect-file-cache`

After some investigation, it seems Zombie is getting gzipped Javascript, which is does not handle. So I ran the requests through [Charles](http://www.charlesproxy.com), and this is what I get. It seems that `connect-file-cache` returns gzipped content in cases where I don't expected it to.

Looking at the source, it seems this line is true regardless of the value of `accept-encoding`
`if cacheHash.gzippedData and req.headers['accept-encoding']?.indexOf /gzip/`

This is the small test case I investigated:

**Basic request**

`curl http://localhost:3000/js/script.js --proxy localhost:8888`

```
HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/javascript
Last-Modified: Wed, 03 Oct 2012 06:02:00 GMT
Content-Length: 2188
Date: Wed, 03 Oct 2012 06:06:10 GMT
Connection: keep-alive
```

**Asking for a gzip response**

`curl http://localhost:3000/js/script.js --proxy localhost:8888 --header 'accept-encoding: gzip'`

```
[same except]
Content-Encoding: gzip
Content-Length: 675
```

**Asking for an untouched response**

`curl http://localhost:3000/js/script.js --proxy localhost:8888 --header 'accept-encoding: identity'`

```
[same except]
Content-Encoding: gzip
Content-Length: 675
```

**Any encoding?**

`curl http://localhost:3000/js/script.js --proxy localhost:8888 --header 'accept-encoding: foobar'`

```
[same except]
Content-Encoding: gzip
Content-Length: 675
```

**Comment by ghost** [comment]
The problem I think is that he was passing a regex to indexOf instead of a string. For me it was fixed both of these ways (either match with regex, or indexOf with string):

``` coffeescript
    if cacheHash.gzippedData and req.headers['accept-encoding']?.match /gzip/
```

``` coffeescript
    if cacheHash.gzippedData and req.headers['accept-encoding']?.indexOf 'gzip'
```

**Comment by TrevorBurnham** [comment]
Closed by #9.
